mirascope/base/ops_utils.py

```diff
@@ -54,17 +53,18 @@ def get_wrapped_client(client: T, self: Union[BaseCall, BaseEmbedder]) -> T:
                 import logfire
 
                 logfire.instrument_openai(client)  # type: ignore
-            if callable(op):
+            elif callable(op):
                 client = op(client)
     return client
 
 
 C = TypeVar("C")
+F = TypeVar("F", bound=Callable[..., Any])
+DecoratorType = Callable[[F], F]
```

Comment:
Please move this to above `mirascope_span` so it's right above the function that consumes it (easier to grok)